src/core/Akka/Event/DeadLetterListener.cs

```diff
@@ -65,112 +59,80 @@ private void IncrementCount()
                 _count++;
             }
         }
-
-        /// <summary>
-        /// TBD
-        /// </summary>
-        /// <param name="message">TBD</param>
-        /// <returns>TBD</returns>
         protected override bool Receive(object message)
         {
             if (_isAlwaysLoggingDeadLetters)
             {
-                return ReceiveWithAlwaysLogging()(message);
+                return ReceiveWithAlwaysLogging(message);
             }
 
             return Context.System.Settings.LogDeadLettersSuspendDuration != Timeout.InfiniteTimeSpan
                 ? ReceiveWithSuspendLogging(Context.System.Settings.LogDeadLettersSuspendDuration)(message)
-                : ReceiveWithMaxCountLogging()(message);
+                : ReceiveWithMaxCountLogging(message);
         }
 
-        private Receive ReceiveWithAlwaysLogging()
+        private bool ReceiveWithAlwaysLogging(object message)
```

Comment:
Minor perf optimization - avoid allocating new behaviors here if we can help it.